elytron/WFCORE-3947-ocsp-ssl-revocation.adoc

```diff
@@ -0,0 +1,81 @@
+= OCSP certificate revocation
+:author:            Martin Mazanek
+:email:             mmazanek@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3947[WFCORE-3947]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1617[ELY-1617]
+
+=== Dev Contacts
+
+* mailto:mmazanek@redhat.com[Martin Mazanek]
+
+=== QE Contacts
+
+* TBD
+
+=== Affected Projects or Components
+
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* New class RevocatingTrustManager
+* Configurable certificate revocation using OCSP. This will be implemented using PKIXRevocationChecker.
+* Configuration will be done using Builder class and will include:
+** Current configuration in X509CRLExtendedTrustManager
+** OCSP Responder URI for overriding default, where the default is retrieved from AIA extension (RFC5280) or from 'ocsp.responderURL' security property
```

Comment:
Speaking about possibility of configuring `ocsp.responderURL`, I think we should make possible to configure also other available JVM properties via our own configuration that are described in [Appendix C: On-Line Certificate Status Protocol (OCSP) Support](https://docs.oracle.com/javase/8/docs/technotes/guides/security/certpath/CertPathProgGuide.html#AppC).

Reply:
This is configurable using security-properties element of elytron subsystem. Or do you want a specific thing of configuring those related to ocsp?

Reply:
I see, still I think it makes sense to put it into the configuration directly under the introduced `ocsp` element. My motivation is driven by following two assumptions:
1. having it directly under the `ocsp` element is easier for user to realize what all can be configured actually and is more readable than in a global set of properties
1. `security-properties` under the `elytron` subsystem as I understand them, means global configuration, although here we are talking about local, per `trust-manager` configuration

Reply:
Sorry, my bad, I thought that these were global, but looking at the code it seems that they are loaded from the properties only at init, so it can be set up differently for each trust-manager. Will change, thanks for pointing this out

Reply:
However, after a second look it is basically implemented, as the API introduced in JDK8 [JEP124](http://openjdk.java.net/jeps/124) overrides these properties. There is a little change, where the `responderCert*` properties are passed as one certificate, so the certificate will be selected by alias instead of subject/issuer name or serial number to isolate it from the keystore, which is more user friendly. So the only reason why to use the properties would be to set up global defaults

Reply:
Just a minor comment, I think it would be nice to give an option to configure the `responderCert`. This is usually something that is always requested by OCSP users (sometimes the certificate used by the responder is not the one that issued the final certificate being checked, and that certificate should be configured). In the java API is the method [setOcspResponderCert](https://docs.oracle.com/javase/8/docs/api/java/security/cert/PKIXRevocationChecker.html#setOcspResponderCert-java.security.cert.X509Certificate-). It could be just the alias in the trust-store configured (at EAP level I mean). For example in apache httpd the option is this one [SSLOCSPResponderCertificateFile](https://httpd.apache.org/docs/2.4/mod/mod_ssl.html#sslocsprespondercertificatefile) It's something that is always exposed by the configuration.

Reply:
Thanks for noticing that, I somehow missed it. It should be included now :)